aiida/tools/data/array/kpoints/legacy.py

```diff
@@ -342,6 +342,10 @@ def _num_points_from_coordinates(path, point_coordinates, kpoint_distance=None):
     explicit_kpoints = [tuple(point_coordinates[path[0][0]])]
     labels = [(0, path[0][0])]
 
+    if not all([_.is_integer() for _ in num_points if isinstance(_, (float, numpy.float64))]):
+        raise ValueError('Could not determine number of points as a whole number. num_points={}'.format(num_points))
+    num_points = [int(_) for _ in num_points]
```

Comment:
I don't see this variable `num_points` being defined in the old code, why do we need to define this now?

Reply:
It is defined in the `if-else` block immediately above.

Reply:
Actually, I think this code will except in the exception, because it references `num_points` in the exception message but it is only defined afterwards. Would highly recommend actually adding a test...

Reply:
The issue here is that the new `numpy` version apparently makes the contents of the list `numpy.float64` types, where it was previously integers.

Reply:
> Actually, I think this code will except in the exception, because it references `num_points` in the exception message but it is only defined afterwards. Would highly recommend actually adding a test...

No? It's defined above. Otherwise the creation of the list for the `all()` check would fail ...

Reply:
...
So all this code is doing, is simply making sure that the contents of the previously created `num_points` are all integers as is expected from this code, apparently.

Reply:
I see, apologies, didn't think about it being defined above. Last question, is there a reason for the `if isinstance(_, (float, numpy.float64))`? Shouldn't they all be integers for it to be valid, i.e. should the following not be sufficient:
```
if any([not _.is_integer() for _ in num_points]):
    raise ValueError('Could not determine number of points as a whole number. num_points={}'.format(num_points))
```

Reply:
> I see, apologies, didn't think about it being defined above.

No worries :)

> Last question, is there a reason for the `if isinstance(_, (float, numpy.float64))`? Shouldn't they all be integers for it to be valid, i.e. should the following not be sufficient:
> 
> ```
> if any([not _.is_integer() for _ in num_points]):
>     raise ValueError('Could not determine number of points as a whole number. num_points={}'.format(num_points))
> ```

I did that originally, but for some of the cases in the `if-else` block above, it correctly creates instances of `int`s, which don't have the `is_integer()` method ... so it failed in the CI and I added the `if`-condition.

In the end, there shouldn't be an issue for this check at all, we could just force cast the type to `int` and be done with it. But as I'm not completely sure what is being done in this code, I'd rather be overly careful than simply force type casting.